kolibri_instant_schools_plugin/assets/src/views/reset-password-page/new-password-form.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+
+  <div>
+    <h1>{{ $tr('resetPasswordHeader') }}</h1>
+
+    <form @submit.prevent="submit">
+      <k-textbox
+        :autofocus="true"
+        ref="newPw"
+        type="password"
+        :label="$tr('newPw')"
+        :disabled="disable"
+        :maxlength="120"
+        :invalid="!newPwIsValid"
+        :invalidText="$tr('required')"
+        @blur="newPwBlurred = true"
+        v-model="newPw"
+      />
+      <k-textbox
+        ref="newPwConfirm"
+        type="password"
+        :label="$tr('newPwConfirm')"
+        :disabled="disable"
+        :maxlength="120"
+        :invalid="!newPwConfirmIsValid"
+        :invalidText="$tr('passwordsDoNotMatch')"
+        @blur="newPwConfirmBlurred = true"
+        v-model="newPwConfirm"
+      />
+
+      <k-button
+        type="submit"
+        :text="$tr('saveButton')"
+        :disabled="disable"
+        :primary="true"
+      />
+    </form>
+  </div>
+
+</template>
+
+
+<script>
+
+  import kTextbox from 'kolibri.coreVue.components.kTextbox';
+  import kButton from 'kolibri.coreVue.components.kButton';
+
+  export default {
+    name: 'newPasswordForm',
+    components: {
+      kButton,
+      kTextbox,
+    },
+    props: {
+      disable: {
+        type: Boolean,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        formSubmitted: false,
+        newPw: '',
+        newPwConfirm: '',
+        newPwBlurred: false,
+        newPwConfirmBlurred: false,
+      };
+    },
+    computed: {
+      newPwIsValid() {
+        if (this.newPwShouldValidate) {
+          return this.newPw !== '';
+        }
+        return true;
+      },
+      newPwConfirmIsValid() {
+        if (this.newPwConfirmShouldValidate) {
+          return this.newPw === this.newPwConfirm;
+        }
+        return true;
+      },
+      newPwShouldValidate() {
+        return this.newPwBlurred || this.formSubmitted;
+      },
+      newPwConfirmShouldValidate() {
+        return (
+          (this.newPw !== '' && this.newPwBlurred && this.newPwConfirm !== '') || this.formSubmitted
```

Comment:
Missing `newPwConfirmBlurred`?

Reply:
Hm, I kind of wanted to avoid invalidating the `newPwConfirm` just for being blank (and thus requiring a different invalidText for that scenario). Here, it only invalidates if it doesn't match the `newPw`. This is covers when `newPwConfirm === ''` and `newPw !== ''`